Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,807 @@
+       Returns:
+           human readable, context, raw response of this command.
+    """
+    ticket_id = args.get('ticket_id')
+    try:
+        response = client.delete_ticket_request(ticket_id)
+    except Exception as e:
+        raise DemistoException(e)
+    if response.get('Result') == 'Success':
+        context = {}
+        old_context = demisto.dt(demisto.context(), f'QuestKace.Ticket(val.ID === {ticket_id})')
+        if old_context:
+            if isinstance(old_context, list):
+                old_context = old_context[0]
+            old_context['IsDeleted'] = True
+            context = {
+                f'QuestKace.Ticket(val.ID === obj.ID)': old_context
+            }
+        return f'Ticket was deleted successfully. Ticket number {ticket_id}', context, {}
+    else:
+        raise DemistoException(f'Error while deleting the ticket.')
+
+
+def fetch_incidents(client: Client, fetch_time: str, fetch_shaping: str, last_run: Dict, fetch_limit: str,
+                    fetch_queue_id: Optional[list] = None, fetch_filter: Optional[str] = None) -> list:
+    """
+    This function will execute each interval (default is 1 minute).
+    Args:
+        client (Client): Quest Kace Client
+        fetch_time: time interval for fetch incidents.
+        fetch_shaping: shaping for the request.
+        fetch_filter: custom filters for the request.
+        fetch_limit: limit for number of fetch incidents per fetch.
+        fetch_queue_id: queue id for fetch, if not given then fetch runs on all tickets in the system
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+    Returns:
+        incidents: Incidents that will be created in Demisto
+    """
+
+    time_format = '%Y-%m-%dT%H:%M:%SZ'
+    if not last_run:  # if first time running
+        new_last_run = {'last_fetch': parse_date_range(fetch_time, date_format=time_format)[0]}
+    else:
+        new_last_run = last_run
+
+    if not fetch_shaping:
+        integration_context = demisto.getIntegrationContext()
+        if integration_context:
+
+            valid_until = integration_context.get('valid_until')
+            time_now = int(time.time())
+            if time_now < valid_until:
+                fetch_shaping = integration_context.get('shaping_fields')
+            else:
+                fetch_shaping = set_shaping(client, fetch_queue_id)
+                integration_context = {
+                    'shaping_fields': fetch_shaping,
+                    'valid_until': int(time.time()) + 3600 * 24
+                }
+                demisto.setIntegrationContext(integration_context)
+        else:
+            fetch_shaping = set_shaping(client, fetch_queue_id)
+            integration_context = {
+                'shaping_fields': fetch_shaping,
+                'valid_until': int(time.time()) + 3600 * 24
+            }
+            demisto.setIntegrationContext(integration_context)
```

Comment:
I think that this logic should move to a separate func and documented further.
Also let's try to think of a way to mock the time changes and test this.